### ISSUE TITLE
Code signing mac builds fails after updating to 1.10.0

### ISSUE THREAD

**Issue by colintremblay-okta** [amplitude/Amplitude-Swift#236]
## Expected Behavior
Code signing shouldn't have changed from 1.9.4 to 1.10.0

## Current Behavior
We're seeing the following error when we code sign our mac application:
```
/PATH_TO_ARCHIVE/PRODUCT_NAME.xcarchive/Products contains invalid products.

Multiple binaries share the same codesign path:
/Applications/PRODUCT_NAME.app/Contents/Frameworks/AnalyticsConnector.framework/Versions/Current

Binaries:
/Applications/PRODUCT_NAME.app/Contents/Frameworks/AnalyticsConnector.framework/Versions/Current/AnalyticsConnector
/Applications/PRODUCT_NAME.app/Contents/Frameworks/AnalyticsConnector.framework/AnalyticsConnector

This can happen if your build process copies frameworks by following symlinks.
```

Downgrading our Amplitude SDK to 1.9.4 resolves our issue.

## Possible Solution

It seems that #235 modified the way `AnalyticsConnector` is pulled in, perhaps this is the culprit? It's also possible that one of the recent changes within the [analytics-connector-ios](https://github.com/amplitude/analytics-connector-ios) library itself could be the cause.

## Steps to Reproduce
1. Create a new macOS project in Xcode (code signing with `Personal Team` is fine)
2. Add Amplitude-Swift as a Swift Package dependency (use `Up to next major version` so it grabs 1.10.0). Make sure to add the dependency to the target.
3. Archive the project (`Product` -> `Archive`)
4. When the `Organizer` window opens, highlight the archive and click the `Validate App` button in the right panel.

The error mentioned above in Current Behavior will pop up.

Next, change the dependency to use `Exact version` and set 1.9.4 as the version. Repeat steps 3 and 4. The error will not appear (if you used `Personal Team` and actually try to complete the validation it will fail, but that's normal).

## Environment
- SDK Version: 1.10.0
- OS Info: Code signing is happening on a mac running macOS 14.1, Xcode: 15.2

**Comment by crleona** [comment]
Hi @colintremblay-okta - it look like we were not preserving symlinks when zipping the frameworks for distribution, this should be fixed as of analytics-connector-ios 1.2.4.

Binaries from this repo will be fixed in https://github.com/amplitude/Amplitude-Swift/pull/237, but they should only be for Carthage users.

**Comment by colintremblay-okta** [comment]
It looks like Swift Package Manager pulls in analytics-connector-ios 1.2.4 automatically now, so everything is good on our end!
